custom_components/opnsense/__init__.py

```
@@ -305,4 +320,13 @@ def _get_opnsense_state_value(self, path, default=None):
 
     def _get_opnsense_client(self) -> OPNsenseClient:
+        if self.hass is None:
+            return None
         return self.hass.data[DOMAIN][self.config_entry.entry_id][OPNSENSE_CLIENT]
+
+    async def async_added_to_hass(self) -> None:
+        await super().async_added_to_hass()
+        if self._client is None:
+            self._client: OPNsenseClient = self._get_opnsense_client()
+        if self._client is None:
+            _LOGGER.error(f"Unable to get client in async_added_to_hass.")
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)